optimum/intel/ipex/modeling_base.py

```diff
@@ -113,21 +114,35 @@ def ipex_jit_trace(model, task, use_cache):
     # Only ipex >= 2.3.0 supports tpp. The tpp is only verified for llm in generation tasks.
     if is_ipex_version(">=", "2.3.0") and task in _IPEX_EXPORTED_GENERATION_TASKS:
         _enable_tpp()
-    model = ipex.optimize(model.eval(), dtype=model.dtype, inplace=True)
-    # Disable repack while jit tracing to reduce the memory
-    ipex._C.disable_jit_linear_repack()
-    with torch.no_grad():
-        trace_model = torch.jit.trace(
-            model,
-            example_kwarg_inputs=sample_inputs,
-            strict=False,
-            check_trace=False,
-        )
-        trace_model = torch.jit.freeze(trace_model)
-        trace_model(**sample_inputs)
-        trace_model(**sample_inputs)
 
-    return trace_model
+    if task == "text2text-generation":
+        if is_torch_version("<", "2.5.0"):
+            warnings.warn(
+                "The Seq2Seq model will keep eager mode, please upgrade your pytorch version for better performance with graph mode."
+            )
+        else:
+            torch._inductor.cpp_wrapper = True
+            model.forward = torch.compile(model.forward, dynamic=True)
+            # compile
+            warnings.warn("Run a few epochs for torch.compile by your use case.")
```

Comment:
So for seq2seq models we would just use torch.compile without applying any IPEX optimizations afterwards ? Also not sure we want to support both TorchScript and torch.compile, would be better to start deprecating TorchScript and use torch.compile for all models

Reply:
Yes, Ella, we are starting to deprecating TorchScript and use torch.compile. We are trending to go the "torch.compile for graph compilation + ipex for peak perf custom op" way to make optimum-intel get peak performance for transformers models on Intel platforms(cpu, xpu). And we'll start from see2seq model and enable more tasks along the way per pytorch & ipex readiness.

Reply:
Do you know when we will be able to use torch.compile for all tasks ? I'd rather wait for us to be able to use torch.compile for all tasks and not have different behavior for different tasks. Also currently there is no IPEX optimization applied

Reply:
Hi Ella, I have tested torch.compile and found it still has many issues. But we cannot block every things by it. I know enabling different tasks with 2 different optimization ways (torch.compile and jit) is not friendly to users because they need to keep 2 knowledge, but most users do not care what kind of optimization you take as long as the performance is great to them. 

In addition, we cannot give up the text2text-generation task only because of the little inconvenience.